frontend/packages/dev-console/src/components/topology2/components/nodes/BaseNode.tsx

```diff
@@ -72,21 +82,23 @@ const BaseNode: React.FC<BaseNodeProps> = ({
   const refs = useCombineRefs<SVGEllipseElement>(hoverRef, dragNodeRef);
 
   React.useLayoutEffect(() => {
-    if (hover) {
-      onShowCreateConnector && onShowCreateConnector();
-    } else {
-      onHideCreateConnector && onHideCreateConnector();
+    if (editAccess) {
+      if (hover) {
+        onShowCreateConnector && onShowCreateConnector();
+      } else {
+        onHideCreateConnector && onHideCreateConnector();
+      }
     }
-  }, [hover, onShowCreateConnector, onHideCreateConnector]);
+  }, [hover, onShowCreateConnector, onHideCreateConnector, editAccess]);
 
   return (
     <g className="odc2-base-node">
       <NodeShadows />
       <g
         data-test-id="base-node-handler"
         onClick={onSelect}
-        onContextMenu={onContextMenu}
-        ref={refs}
+        {...(editAccess ? { onContextMenu } : {})}
```

Comment:
This isn't right. Context menu actions already have their own access checks built in.

Reply:
This is for hiding the ```contextMenu```. Because we are hiding all the option which user has no access to.

Reply:
I guess we don't have any context menu actions that don't act directly on the resource itself, so it's probably ok to hide it.
However:
```suggestion
        onContextMenu={editAccess && onContextMenu}
```

Reply:
Got this warning when I change it to ```onContextMenu={editAccess && onContextMenu}```

![Screenshot from 2019-11-24 09-47-31](https://user-images.githubusercontent.com/2561818/69489669-c6706680-0ea1-11ea-9a9c-094c27010f29.png)

Reply:
How about this then.
The other way just doesn't seem very clear but maybe it's just me. @andrewballantyne @rohitkrai03  what do you suggest?
```suggestion
        onContextMenu={editAccess && onContextMenu || null}
```

Reply:
Not really, ternaries are the only other option... but it's essentially the same thing. But it might be cleaner to read:
```suggestion
        onContextMenu={editAccess ? onContextMenu : null}
```
🤷‍♂